sql-queries-3/sql-alter-drop-if-exist/drop-conditional-alter-mysql.sql

```diff
@@ -0,0 +1,29 @@
+DELIMITER //
+
+CREATE PROCEDURE DropColumnIfExists()
+BEGIN
+    DECLARE column_exists INT DEFAULT 0;
+
+    -- Check if the column exists
+    SELECT COUNT(*) INTO column_exists 
+    FROM information_schema.columns 
+    WHERE table_schema = 'University' 
+      AND table_name = 'Student' 
+      AND column_name = 'national_id';
+
+    -- If the column exists, drop it
+    IF column_exists > 0 THEN
+        SET @sql = 'ALTER TABLE Student DROP COLUMN national_id';
+        PREPARE stmt FROM @sql;
+        EXECUTE stmt;
+        DEALLOCATE PREPARE stmt;
+    END IF;
```

Comment:
In my daily experience with MySQL stored procedures, the prepared statement isn't necessary unless you have a parameterized SQL statement to run. So, in this simple use case, you could just write:

`IF column_exists > 0 THEN
    ALTER TABLE Student DROP COLUMN national_id;
END IF;
`

I'm not familiar enough with PostgreSQL or SQL Server stored procs to know whether you can achieve similar results.